File: angular-ngrx/src/app/models/book.ts
import { Author } from './author';
import { Rating } from './rating';
import { Editor } from './editor';

export class Book {

  isbn: string;
  title: string;
  author: Author;
  editor: Editor;
  description: string;
  page: string;
  price: string;
  rating: Rating;

  // tslint:disable-next-line:max-line-length
  constructor(isbn: string, title: string, author: Author, editor: Editor, description: string, page: string, price: string, rating: Rating) {
    this.isbn = isbn;
    this.title = title;
    this.author = author;
    this.description = description;
    this.price = price;
    this.rating = rating;
  }

}

File: book-microservice/src/main/java/com/kali/sample/dto/Book.java
package com.kali.sample.dto;

public class Book extends Dto {

	private String isbn;
	private String title;
	private Author author;
	private Editor editor;
	private String description;
	private String page;
	private double price;
	private int rating;

	public Book() {
		super();
	}

	public Book(String isbn, String title, String description, String page, double price) {
		super();
		this.isbn = isbn;
		this.title = title;
		this.description = description;
		this.page = page;
		this.price = price;
	}

	public String getIsbn() {
		return isbn;
	}

	public void setIsbn(String isbn) {
		this.isbn = isbn;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public Author getAuthor() {
		return author;
	}

	public void setAuthor(Author author) {
		this.author = author;
	}

	public Editor getEditor() {
		return editor;
	}

	public void setEditor(Editor editor) {
		this.editor = editor;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getPage() {
		return page;
	}

	public void setPage(String page) {
		this.page = page;
	}

	public double getPrice() {
		return price;
	}

	public void setPrice(double price) {
		this.price = price;
	}

	public int getRating() {
		return rating;
	}

	public void setRating(int rating) {
		this.rating = rating;
	}

}
